Sources/ApiVideoPlayer/Views/ControlsView.swift

```diff
@@ -295,31 +304,69 @@ extension ControlsView: ApiVideoPlayerControllerPlayerDelegate {
 }
 
 extension ControlsView: ActionBarViewDelegate {
-    private func toggleSubtitleView(subtitleButton: UIButton) {
+    private func toggleSubtitleView(button: UIButton, buttonType: ButtonType) {
         timer.clear()
-        let posX = subtitleButton.frame.origin.x + 6
+        let posX = button.frame.origin.x + 6
         let posY = self.frame.height - 90
 
-        if let subtitleView = subtitleView,
-           subtitleView.isDescendant(of: self)
-        {
-            removeSubtitleView()
-        } else {
-            let notOptionalSubtitleView = SubtitleListView(
-                frame: CGRect(x: posX, y: posY, width: 130, height: 3 * 45),
-                locales: playerController.subtitleLocales,
-                selectedLocale: playerController.currentSubtitleLocale
-            )
-            notOptionalSubtitleView.delegate = self
-            addSubview(notOptionalSubtitleView)
-            bringSubviewToFront(notOptionalSubtitleView)
-            subtitleView = notOptionalSubtitleView
+        switch buttonType {
```

Comment:
This looks like duplicated code. It is not good for maintenance and readibility.
This should be factorized.